apps/challenges/aws_utils.py

```diff
@@ -668,25 +675,39 @@ def register_task_def_by_challenge_pk(client, queue_name, challenge):
             # challenge host auth token to be used by code-upload-worker
             token = JwtToken.objects.get(user=challenge.creator.created_by)
             if challenge.is_static_dataset_code_upload:
-                definition = task_definition_static_code_upload_worker.format(
+                code_upload_container = (
+                    code_upload_container_defination.format(
+                        queue_name=queue_name,
+                        code_upload_container_name=code_upload_container_name,
+                        auth_token=token.refresh_token,
+                        cluster_name=cluster_name,
+                        cluster_endpoint=cluster_endpoint,
+                        certificate=cluster_certificate,
+                        log_group_name=log_group_name,
+                        EVALAI_DNS=EVALAI_DNS,
+                        EFS_ID=efs_id,
+                        **COMMON_SETTINGS_DICT,
+                        **challenge_aws_keys,
+                    )
+                )
+                submission_container = submission_container_defination.format(
                     queue_name=queue_name,
                     container_name=container_name,
-                    code_upload_container_name=code_upload_container_name,
                     ENV=ENV,
                     challenge_pk=challenge.pk,
-                    auth_token=token.refresh_token,
-                    cluster_name=cluster_name,
-                    cluster_endpoint=cluster_endpoint,
-                    certificate=cluster_certificate,
-                    CPU=worker_cpu_cores,
-                    MEMORY=worker_memory,
                     log_group_name=log_group_name,
-                    EVALAI_DNS=EVALAI_DNS,
-                    EFS_ID=efs_id,
                     AWS_SES_REGION_NAME=AWS_SES_REGION_NAME,
                     AWS_SES_REGION_ENDPOINT=AWS_SES_REGION_ENDPOINT,
                     **COMMON_SETTINGS_DICT,
-                    **challenge_aws_keys,
+                    **aws_keys,
+                )
+                definition = task_definition_static_code_upload_worker.format(
+                    queue_name=queue_name,
+                    code_upload_container=code_upload_container,
+                    submission_container=submission_container,
+                    CPU=worker_cpu_cores,
+                    MEMORY=worker_memory,
+                    **COMMON_SETTINGS_DICT,
```

Comment:
Please move this to a separate method. This piece of code is unreadable. Also, why do we need to substitute same variables in two task definitions can't we dedup it and make this code more readble?

Reply:
Sure, we can move that to a different function, the common variables are `queue_name` and `COMMON_SETTINGS_DICT`, but still since `code_upload_container` and `submission_container_defination` are passed as a variable to `task_definition_static_code_upload_worker` so we need to format the all the varaibles of them before right? before passing

Reply:
@Ram81, Should we test these changes first on fargate and if everything works we can maybe add the container definitions in  task_defination and task_definations_code_upload_worker as well and create three functions on `get_challenge_task_defn`, `get_code_upload_challenge_task_defn`, `get_static_dataset_code_upload_challenge_task_defn` in `task_definations.py` and eventually this peice of code can shift to `get_static_dataset_code_upload_challenge_task_defn` function.